### ISSUE TITLE
Crash on Java 8

### ISSUE THREAD

**Issue by phgn0** [georgewfraser/java-language-server#132]
Hi 👋 

I'm working inside a large monorepo which currently uses Java 8 - thus `JAVA_HOME` points to JDK 8.
Building the vscode extension (using JDK 13) with the change from #129 **causes the LSP server to crash** during runtime. See the stacktrace below.

My guess is the LSP server does not work with Java 8, maybe because of changes in the Compiler API. The LSP version `v0.2.31` (before the change) still works with Java 8, because the bundled JDK is used always.

You've commented about this topic in #124, but do you see a way to use the bundled JDK 13 for the Java Compiler API, and use the version from `JAVA_HOME` to get code completion?

Otherwise it would be great to have a fallback to the bundled JDK 13 if the user-provided JDK is too old to support the LSP server.

```
14:58:49.659	SEVERE	LSP connect	java.lang.IllegalArgumentException: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
java.lang.RuntimeException: java.lang.IllegalArgumentException: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
	at jdk.compiler/com.sun.tools.javac.api.JavacTool.getTask(JavacTool.java:198)
	at org.javacs.ReusableCompiler.getTask(ReusableCompiler.java:123)
	at org.javacs.CompileBatch.batchTask(CompileBatch.java:106)
	at org.javacs.CompileBatch.<init>(CompileBatch.java:31)
	at org.javacs.JavaCompilerService.doCompile(JavaCompilerService.java:76)
	at org.javacs.JavaCompilerService.loadCompile(JavaCompilerService.java:67)
	at org.javacs.JavaCompilerService.compileBatch(JavaCompilerService.java:93)
	at org.javacs.JavaCompilerService.compile(JavaCompilerService.java:349)
	at org.javacs.JavaCompilerService.compile(JavaCompilerService.java:344)
	at org.javacs.action.CodeActionProvider.codeActionsForCursor(CodeActionProvider.java:37)
	at org.javacs.JavaLanguageServer.codeAction(JavaLanguageServer.java:506)
	at org.javacs.lsp.LSP.connect(LSP.java:400)
	at org.javacs.Main.main(Main.java:30)
Caused by: java.lang.IllegalArgumentException: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
	at jdk.compiler/com.sun.tools.javac.file.Locations$SystemModulesLocationHandler.update(Locations.java:1905)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SystemModulesLocationHandler.handleOption(Locations.java:1850)
	at jdk.compiler/com.sun.tools.javac.file.Locations.handleOption(Locations.java:2114)
	at jdk.compiler/com.sun.tools.javac.file.BaseFileManager.handleOption(BaseFileManager.java:269)
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.handleOption(JavacFileManager.java:1337)
	at jdk.compiler/com.sun.tools.javac.file.BaseFileManager$2.handleFileManagerOption(BaseFileManager.java:222)
	at jdk.compiler/com.sun.tools.javac.main.Option.process(Option.java:1205)
	at jdk.compiler/com.sun.tools.javac.main.Option.handleOption(Option.java:1153)
	at jdk.compiler/com.sun.tools.javac.file.BaseFileManager.handleOption(BaseFileManager.java:232)
	at java.compiler@13.0.1/javax.tools.ForwardingJavaFileManager.handleOption(ForwardingJavaFileManager.java:99)
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedJavaFileManager.handleOption(ClientCodeWrapper.java:266)
	at jdk.compiler/com.sun.tools.javac.main.Arguments.doProcessArgs(Arguments.java:390)
	at jdk.compiler/com.sun.tools.javac.main.Arguments.processArgs(Arguments.java:347)
	at jdk.compiler/com.sun.tools.javac.main.Arguments.init(Arguments.java:246)
	at jdk.compiler/com.sun.tools.javac.api.JavacTool.getTask(JavacTool.java:185)
	... 12 more

[Error - 2:58:49 PM] Request textDocument/codeAction failed.
  Message: java.lang.IllegalArgumentException: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
  Code: -32603 
14:58:49.661	INFO	Parser parseJavaFileObject	...using cached parse
14:58:49.666	INFO	Parser parseJavaFileObject	...using cached parse
14:58:49.873	INFO	JavaLanguageServer lint	Lint 1 files...
14:58:49.875	INFO	CompileBatch options	JDIR: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
14:58:49.875	SEVERE	Main main	Compiler is already in-use!
java.lang.RuntimeException: Compiler is already in-use!
	at org.javacs.ReusableCompiler.getTask(ReusableCompiler.java:111)
	at org.javacs.CompileBatch.batchTask(CompileBatch.java:106)
	at org.javacs.CompileBatch.<init>(CompileBatch.java:31)
	at org.javacs.JavaCompilerService.doCompile(JavaCompilerService.java:76)
	at org.javacs.JavaCompilerService.loadCompile(JavaCompilerService.java:67)
	at org.javacs.JavaCompilerService.compileBatch(JavaCompilerService.java:93)
	at org.javacs.JavaCompilerService.compile(JavaCompilerService.java:349)
	at org.javacs.JavaCompilerService.compile(JavaCompilerService.java:344)
	at org.javacs.JavaLanguageServer.lint(JavaLanguageServer.java:61)
	at org.javacs.JavaLanguageServer.doAsyncWork(JavaLanguageServer.java:523)
	at org.javacs.lsp.LSP.connect(LSP.java:247)
	at org.javacs.Main.main(Main.java:30)
```

**Comment by phgn0** [comment]
I just saw you wanted to revert using the system SDK (causing my issue) in #130. 
The change didn't went through, as the state of master still includes the change that breaks the LSP for Java 8.

I created a new PR with the change in #133.

**Comment by georgewfraser** [comment]
This is fixed now, right?

**Comment by phgn0** [comment]
Yes this should be fixed with #133.
